pkg/graph/transforms.go

```diff
@@ -33,6 +38,82 @@ func (nt NoTransform) Name() string {
 	return "no-transform"
 }
 
+type AttributesFilterTransform struct {
+	Filter *eventingv1.TriggerFilter
+}
+
+var _ Transform = &AttributesFilterTransform{}
+
+func (aft *AttributesFilterTransform) Apply(et *eventingv1beta3.EventType, tfc TransformFunctionContext) (*eventingv1beta3.EventType, TransformFunctionContext) {
+	etAttributes := make(map[string]*eventingv1beta3.EventAttributeDefinition)
+	for i := range et.Spec.Attributes {
+		etAttributes[et.Spec.Attributes[i].Name] = &et.Spec.Attributes[i]
+	}
+
+	for k, v := range aft.Filter.Attributes {
+		if attribute, ok := etAttributes[k]; ok {
+			if attribute.Value != v {
+				regexp, err := buildRegexForAttribute(attribute.Value)
+				if err != nil {
+					return nil, tfc
+				}
+
+				if regexp.MatchString(v) {
+					attribute.Value = v
+				} else {
+					return nil, tfc
+				}
+			}
+		} else {
+			etAttributes[k] = &eventingv1beta3.EventAttributeDefinition{
+				Name:     k,
+				Value:    v,
+				Required: true,
+			}
+		}
+	}
+
+	updatedAttributes := make([]eventingv1beta3.EventAttributeDefinition, 0, len(et.Spec.Attributes))
+	for _, v := range etAttributes {
+		updatedAttributes = append(updatedAttributes, *v)
+	}
+
+	et.Spec.Attributes = updatedAttributes
+
+	return et, tfc
+}
+
+func (aft *AttributesFilterTransform) Name() string {
+	return "attributes-filter"
+}
+
+func buildRegexForAttribute(attribute string) (*regexp.Regexp, error) {
+	chunks := []string{"^"}
+
+	var chunk strings.Builder
+	for i := 0; i < len(attribute); {
+		if attribute[i] == '{' {
+			chunks = append(chunks, chunk.String(), "[a-zA-Z]+")
+			chunk.Reset()
+
+			offset := strings.Index(attribute[i:], "}")
+			if offset == -1 {
+				return nil, fmt.Errorf("no closing bracket for variable")
+			}
+
+			i += offset + 1
+			continue
+		}
+
+		chunk.WriteByte(attribute[i])
+		i++
+	}
+
+	chunks = append(chunks, chunk.String(), "$")
```

Comment:
are we validating that attributes with variables are well formatted?

I think this would produce wrong results when the value looks like this one `{abc}}`